### DISCUSSION TITLE
How to calculate the distance between two satellites and predict the probability of a possible collision

### DISCUSSION THREAD

**Discussion by leonbusy** [skyfielders/python-skyfield#768] [Q&A]
Hi, I wish you all the best.
I am recently working on using python and the sgp4 algorithm to calculate the distance between two satellites and possibly predict the probability of a collision between them, but I don't know how to implement my idea, can you give me some advice, thanks.

**Comment by davidmikolas** [comment]
if you get positions for two satellites using their TLEs at the same times, you can get the distance between them (fairly accurately) by simply subtracting, squaring, summing over x, y, z and taking the square root. 

But remember TLEs and SGP4 are only accurate to a few kilometers at most, so while potentially dangerous (parts per million) situations can be flagged using them, real predictions of collisions use much more accurate models that aren't readily available to the general public.

You can read more about potential collisions which in spaceflight words are called "conjunctions" here: https://celestrak.org/SOCRATES/ and you can use those predictions to check your own calculations

All of that said, _**predicting collisions is fun!**_ So here are some links in Space Exploration Stack Exchange you might find interesting:

 - https://space.stackexchange.com/questions/24488/how-might-one-approach-using-ai-convolutional-neural-network-to-predict-collis (includes a video)
 -  https://space.stackexchange.com/questions/48021/what-was-the-relative-velocity-of-kosmos-2004-and-the-changzheng-rocket-stage-at
 - https://space.stackexchange.com/questions/29946/algorithmic-methods-or-techniques-to-find-conjunctions-in-high-n-keplerian-eleme
 - https://space.stackexchange.com/questions/57352/closest-approach-between-the-jwst-launch-vehicle-and-santa-clauss-sleigh
 - https://space.stackexchange.com/questions/33161/algorithmic-methods-or-techniques-to-find-conjunctions-in-large-ensembles-of-sta
 - https://space.stackexchange.com/questions/41109/what-does-msr-represent-in-the-context-of-this-predicted-satellite-conjunction

**Reply by leonbusy** [reply]
Hi, David mikolas,
Thank you for your prompt reply, this will give me a lot of good ideas. I would also like to ask where I can find the python code for orbital proximity prediction for SGP4, I have searched many sites and have not found anything of reference value, which has caused me a lot of frustration. Many thanks again！
